### ISSUE TITLE
Validation Data

### ISSUE THREAD

**Issue by jmrichardson** [kingychiu/target-permutation-importances#25]
Hi, it doesn't appear that your code uses a validation set?  Is that by design? In other words why would you use the training set for the predictions?

**Comment by jmrichardson** [comment]
Sorry misunderstood the process.  All good.

**Comment by kingychiu** [comment]
Hi @jmrichardson, sorry for the confusion.

The feature selection example here:
- setup a train-test split
- compute importance on the train set
- select feature on the train set
- validate the performance on the test set

https://www.kaggle.com/code/kingychiu/feature-selection-for-binary-classification-rf?scriptVersionId=146983062


The idea here is you can compute importance with any dataset, like you can fit a linear regression to all your data points: this is more like analysing data (not predicting / machine learning)

For prediction / machine learning, you are right that we need to set up train-test or even train-valid-test to validate the method.

Hope it helps.
